src/RequestFactory.php

```diff
@@ -54,6 +54,35 @@ public function create($serviceKey, $requestKey, array $parameters = [])
     }
 
     /**
+     * Builds a request based on the service and request name, using provided $jsonSerializable as request body
+     *
+     * @param string            $serviceKey
+     * @param string            $requestKey
+     * @param \JsonSerializable $jsonSerializableBody
+     * @return mixed
+     */
+    public function createFromJson($serviceKey, $requestKey, \JsonSerializable $jsonSerializableBody) {
```

Comment:
With `create` you can also pass an array of params, which later can be transformed to JSON if the Content-Type is `application/json`. True. But `DomainEvent` implements `JsonSerializable`,  so I thought it would make little sense to transform it to associative array only to have it serialized again to JSON. Since that method.

No, it is not replacing anything in the body, because the JSON passed is the whole body. For `create` you have a request schema(s) defined per request name with placeholders, which are replaced by values passed in the parameters array. Here there is no need. More over for events the schema cannot be defined as for every event the content of `body` can be different.

I still replace the placeholders in URI/headers though.